infra/gcp/ensure-main-project.sh

```diff
@@ -57,6 +57,11 @@ CLUSTER_USERS_GROUP="gke-security-groups@kubernetes.io"
 # The DNS admins group.
 DNS_GROUP="k8s-infra-dns-admins@kubernetes.io"
 
+# Buckets for the logs of prow
+PROW_BUCKETS=(
+    k8s-prow-infra-logs
```

Comment:
Assuming this is supposed to be the community equivalent of `kubernetes-jenkins`?
```suggestion
    k8s-infra-prow-results
```

Naming is hard when I overthink things.
- First thought: `k8s-infra-prow-logs` to get the consistent `k8s-infra` prefix
- But...
  - "prow logs" sounds like logs of the prow components (plank, sinker, crier, etc)
  - there are more than "logs" that end up `kubernetes-jenkins`, there are also "artifacts" like junit.xml files, coverage files, metadata, etc.
- OK, so `k8s-infra-prow-artifacts` for "artifacts" produced by prow, including job results
- But...
  - "artifacts" clashes with the sig-release meaning of "binaries and container images that represent a given release of a given subproject"
  - makes me thing it's going to be the binaries/images for prow itself, a.k.a. the equivalent to `gcr.io/k8s-prow`
- OK, so how about just `k8s-infra-prow` and be done with it
- But...
  - If we're doing things over, one of the problems with `kubernetes-jenkins` is that it allows per-object ACLs. They allowed that single bucket to be used for multiple purposes, including occasionally storing secrets or credentials.  
  - We don't want per-object ACLs in kubernetes.io, as they are painful and laborious to maintain, audit and enforce.  We are planning on setting an org policy to explicitly require per-bucket IAM only (aka uniform bucket level access).  
  - Thus we should name this bucket for the main purpose everyone expects of `kubernetes-jenkins`, public read access to things about and created by prow jobs for the kubernetes project.  If other purposes arise, they should result in other buckets named/IAM'ed as such.
- OK, so how about `k8s-infra-prow-results`, as in the "results" of prow job runs
  - I looked around in prow's codebase and docs and couldn't find a consistent term for "things that are uploaded to GCS or S3 by the pod-utilities `sidecar` component and `crier`".  Results isn't just "test results" but from the concept of running a prow job and then looking at the "resulting" files in GCS/S3.
  - I considered "data" but that sounds like "training data" or other metadata that is loaded in to help run a job, where I believe this bucket's intended purpose is storage of things about and created by prow jobs

WDYT about `k8s-infra-prow-results`?

Reply:
> "artifacts" clashes with the sig-release meaning of "binaries and container images that represent a given release of a given subproject"

The buckets are prefixed (`k8s-infra-prow`) with a reference. I don't think we should have confusion with other projects. 

Let's go with `k8s-infra-prow-results`, it's more aligned with the needs of this instance.